Source/Magick.NET/Shared/MagickImage.cs

```diff
@@ -4624,10 +4624,35 @@ public void Ping(byte[] data)
         /// <param name="readSettings">The settings to use when reading the image.</param>
         /// <exception cref="MagickException">Thrown when an error is raised by ImageMagick.</exception>
         public void Ping(byte[] data, MagickReadSettings readSettings)
+        {
+            Ping(data, 0, data.Length, readSettings);
```

Comment:
This needs a check to make sure that data is not null. You can check how that is done in the `Read` method.

Reply:
No it doesn't :) There's a check in the final overload

Reply:
There actually appear to be some tests that are failing due to this statement now. When `data` is null you will get a `NullReferenceException` for the `data.Length` statement.

Reply:
Okay, I'll look into it.